app/uk/gov/hmrc/transitmovements/services/XmlParsers.scala

```diff
@@ -32,13 +32,18 @@ import java.time.format.DateTimeParseException
 
 object XmlParsers extends XmlParsingServiceHelpers {
 
-  def movementEORINumberExtractor(rootNode: String, eori: String): Flow[ParseEvent, ParseResult[EORINumber], NotUsed] =
+  def movementEORINumberExtractor(rootNode: String, eori: String): Flow[ParseEvent, ParseResult[Option[EORINumber]], NotUsed] =
     XmlParsing
       .subtree(rootNode :: eori :: "identificationNumber" :: Nil)
       .collect {
         case element if element.getTextContent.nonEmpty => EORINumber(element.getTextContent)
       }
       .single("identificationNumber")
```

Comment:
`single` is an extension method. Could we just create another method entitled `singleOption` and use that, rather than having to match after it?

Reply:
Yes that'd be better! will do